packages/protocol/contracts/layer1/preconf/impl/PreconfWhitelist.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "../iface/IPreconfWhitelist.sol";
+import "../libs/LibPreconfUtils.sol";
+import "../libs/LibPreconfConstants.sol";
+import "src/shared/libs/LibStrings.sol";
+import "src/shared/common/EssentialContract.sol";
+
+/// @title PreconfWhitelist
+/// @custom:security-contact security@taiko.xyz
+contract PreconfWhitelist is EssentialContract, IPreconfWhitelist {
+    // Tracks the total number of operators in the whitelist
+    uint256 public operatorCount;
+
+    // Maps operator index to their corresponding operator addresses
+    mapping(uint256 operatorIndex => address operator) public operatorIndexToOperator;
+
+    function init(address _owner, address _sharedResolver) external initializer {
+        __Essential_init(_owner, _sharedResolver);
+    }
+
+    /// @inheritdoc IPreconfWhitelist
+    function addOperator(address _operatorAddress)
+        external
+        onlyFromOwnerOrNamed(LibStrings.B_PRECONF_WHITELIST_OWNER)
+    {
+        // For simplicity, we assume that the whitelist owner does not add the same operator more
+        // than once.
+        operatorIndexToOperator[operatorCount] = _operatorAddress;
+        unchecked {
+            ++operatorCount;
+        }
+
+        emit OperatorAdded(_operatorAddress);
+    }
+
+    /// @inheritdoc IPreconfWhitelist
+    function removeOperator(uint256 _operatorIndex)
+        external
+        onlyFromOwnerOrNamed(LibStrings.B_PRECONF_WHITELIST_OWNER)
+    {
+        uint256 _operatorCount = operatorCount;
+        require(_operatorIndex < _operatorCount, InvalidOperatorIndex());
+
+        address removedOperator = operatorIndexToOperator[_operatorIndex];
+
+        unchecked {
+            // Bring the last operator to this operator's index
+            address lastOperator = operatorIndexToOperator[_operatorCount - 1];
+            operatorIndexToOperator[_operatorIndex] = lastOperator;
+
+            // Remove the now duplicate operator at the last index
+            delete operatorIndexToOperator[_operatorCount - 1];
```

Comment:
Maybe we can delete this `delete` statement so removing one then adding one will end up wth a non-zero-to-non-zero SSTORE op?